Modules/_io/bufferedio.c

```diff
@@ -1899,9 +1899,9 @@ _bufferedwriter_flush_unlocked(buffered *self)
             goto error;
     }
 
-    _bufferedwriter_reset_buf(self);
 
 end:
+    _bufferedwriter_reset_buf(self);
```

Comment:
I'm a bit skeptical here. Shouldn't the fix go in `buffered_flush_and_rewind_unlocked` instead?

Reply:
I thought when we are flushing out all dirty writes, we should reset `write_end`. This function is called in `write`, `seek` and in `flush_and_rewind`. It's usage appears ok to me in all three cases.

I'm just reading through code and trying to understand the behaviour. So I might have very well misunderstood something. Please correct me in that case.

Reply:
Yes, you may well be right... I have not touched this code in a long time, so I prefer to ask.

A couple of things now:
* can you add a comment here pointing to the issue number and explaning why this is necessary?
* unrelated, but there's a comment at the beginning of `_bufferedwriter_flush_unlocked` mentioning `restore_pos`; that comment is obsolete, so it would be nice to remove it :-)
* in `_io_BufferedWriter_write_impl`, there's a fix for issue `#6629` which might have become unnecessary with this PR; could you check it?